distributed/scheduler.py

```diff
@@ -657,134 +778,277 @@ class TaskState:
     :      The group of tasks to which this one belongs.
     """
 
+    _key: str
+    _hash: Py_hash_t
+    _prefix: object
+    _run_spec: object
+    _priority: tuple
+    _state: str
+    _dependencies: set
+    _dependents: set
+    _has_lost_dependencies: bool
+    _waiting_on: set
+    _waiters: set
+    _who_wants: set
+    _who_has: set
+    _processing_on: WorkerState
+    _retries: Py_ssize_t
+    _nbytes: Py_ssize_t
+    _type: str
+    _exception: object
+    _traceback: object
+    _exception_blame: object
+    _suspicious: Py_ssize_t
+    _host_restrictions: set
+    _worker_restrictions: set
+    _resource_restrictions: dict
+    _loose_restrictions: bool
+    _metadata: dict
+    _actor: bool
+    _group: object
```

Comment:
TaskGroup maybe

Reply:
Same story as with `TaskPrefix`. Since we haven't Cythonized that, it won't have any effect. Also we run into issues as it hasn't been defined yet at this point in the file.

Reply:
I went ahead and put this in quotes, which avoids definition ordering issue. It's still treated as `object` for now though.